lib/Client.php

```diff
@@ -375,9 +354,29 @@ protected function commitSession() {
 	 * @throws OpenIDConnectClientException
 	 */
 	protected function fetchURL($url, $post_body = null, $headers = []) {
-		// TODO: see how to use ownCloud HttpClient ....
+		$this->logger->debug("Fetching URL: $url");
 		try {
-			return parent::fetchURL($url, $post_body, $headers);
+			$client = $this->clientService->newClient();
+			if ($post_body === null) {
+				return $client->get($url, [
+					'headers' => $headers,
+				])->getBody();
+			}
+			// Default content type is form encoded
+			$content_type = 'application/x-www-form-urlencoded';
+
+			// Determine if this is a JSON payload and add the appropriate content type
+			if (\is_object(json_decode($post_body))) {
+				$content_type = 'application/json';
+			}
+
+			// Add POST-specific headers
+			$headers[] = "Content-Type: {$content_type}";
+
+			return $client->post($url, [
```

Comment:
This was the issue why it was reverted back then - fetchURL is used for POST as well as GET